src/osp/Resource/resources.cpp

```diff
@@ -44,16 +44,17 @@ ResId Resources::create(ResTypeId typeId, PkgId pkgId, std::string_view name)
     rPkgType.m_owned.set(std::size_t(newResId));
 
     // Track name
+    auto const& [newIt, success] = rPkgType.m_nameToResId.emplace(std::move(name), newResId);
+    auto const& [key, value]     = *newIt;
+    assert(success); // emplace should always succeed
+
     rPerResType.m_resNames.resize(rPerResType.m_resIds.capacity());
-    SharedString& rResName = rPerResType.m_resNames[std::size_t(newResId)];
-    rResName = SharedString::create(name); // allocates name
-    [[maybe_unused]] auto newIt = rPkgType.m_nameToResId.emplace(rResName, newResId);
-    assert(newIt.second); // emplace should always succeed
+    rPerResType.m_resNames[std::size_t(value)] = key;
```

Comment:
I feel that m_resNames may not have to exist. Names can be stored entirely in PerPkg. There eventually needs to be a way to get which PkgId a ResId belongs to, and be able to obtain the name that way.

Reply:
Certainly, but a bit outside what i was trying to do.